qutip/ui/progressbar.py

```diff
@@ -142,3 +153,39 @@ def update(self, n):
     def finished(self):
         self.t_done = time.time()
         print("\r", "Total run time: %s" % self.time_elapsed())
+
+
+class TqdmProgressBar(BaseProgressBar):
+    """
+    A progress bar using tqdm module
+    """
+
+    def __init__(self, iterations=0, chunk_size=10):
+        from tqdm import tqdm
```

Comment:
We could use the conditional import of tqdm as `from tqdm.auto import tqdm´ (https://github.com/tqdm/tqdm/blob/master/tqdm/auto.py). This seems to be well suited for switching between notebooks or command line code.